server/controllers/userController.js

```diff
@@ -83,6 +85,59 @@ class UserController {
     await addToBlacklist(token);
     return successStat(res, 204, 'message', 'No Content');
   }
+
+  /**
+* @static
+* @description Allows a user to sign in with social accounts
+* @param {Object} req - Request object
+* @param {Object} res - Response object
+* @param {function} next next function to be called
+* @returns {Object} object containing user data and access Token
+* @memberof UserController
+*/
+  static async socialSignin(req, res) {
+    if (!req.user) {
+      return errorStat(res, 404, 'Account not found');
+    }
+
+    // eslint-disable-next-line no-underscore-dangle
+    const userDetails = req.user._json;
+
+    const firstname = userDetails.name.split(' ')[0];
+    const lastname = userDetails.name.split(' ')[1];
+    const username = userDetails.email;
+    const imageUrl = userDetails.picture;
+    const isVerified = userDetails.email_verified;
+    const { email } = userDetails;
+
+    const newUser = await User.findOrCreate({
+      where: { email },
+      defaults: {
+        firstname,
+        lastname,
+        email,
+        password: 'null',
+        bio: '',
+        username,
+        image: imageUrl,
+        verified: isVerified,
+      }
+    });
+    const token = generateToken({
+      id: newUser.id,
+      email: userDetails.email
+    });
+
+    const { bio } = newUser[0];
+
+    return successStat(res, 200, 'user', {
+      email,
```

Comment:
Great Job with the social login @obayomi96. And thanks for the effort you put in to get this delivered. Just a suggestion, since the `signup` and `login` methods return the `firstname` and `lastname`, why don't you mirror it here for uniformity?

Reply:
Thanks for the suggestion @chuxmykel  I did that initially, however, I realized the response spec on the Readme specifically asks for just the info I returned so I had to make changes.